jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -251,15 +392,36 @@ private static FileEntriesLayer addDirectoryContentsToLayer(
       AbsoluteUnixPath basePathInContainer)
       throws IOException {
     FileEntriesLayer.Builder builder = FileEntriesLayer.builder().setName(layerName);
+    HashSet<Path> addedPaths = new HashSet<>();
     new DirectoryWalker(sourceRoot)
         .filterRoot()
-        .filter(path -> Files.isDirectory(path) || pathFilter.test(path))
+        .filter(path -> pathFilter.test(path))
```

Comment:
I vaguely remember I added `Files.isDirectory(path)` to not miss adding empty directories. (It's not because we wanted to add all the directories leading to `/`. Maybe missing empty directories doesn't matter in practice almost all the time, especially for `jib jar`. I wonder if we can just get rid of `Files.isDirectory(path)` and the new `while` loop below? As long as it works, we should be good?

Reply:
Good idea - Tried adding the path directly without adding any empty directories and it seemed to work! Will go with this for now.